internal/downloader/downloader.go

```diff
@@ -59,17 +64,27 @@ func Download(ctx context.Context, destDir string, sourceUrl string, showMsg boo
 		fmt.Println(msg)
 	}
 
-	if d, ok := ctx.Value(downloadImplKey).(downloadImpl); ok {
-		err = d.Download(ctx, destDir, []string{sourceUrl})
-	} else {
+	dl := func(ctx context.Context, sourceUrl, destDir string) error {
 		// conftest's Download function leverages oras under the hood to fetch from OCI. It uses the
 		// global oras client and sets the user agent to "conftest". This is not a thread safe
 		// operation. Here we get around this limitation by ensuring a single download happens at a
 		// time.
 		dlMutex.Lock()
-		err = downloader.Download(ctx, destDir, []string{sourceUrl})
-		dlMutex.Unlock()
+		defer dlMutex.Unlock()
+		return downloader.Download(ctx, destDir, []string{sourceUrl})
+	}
+
+	if d, ok := ctx.Value(downloadImplKey).(downloadImpl); ok {
```

Comment:
If I remembered right, we have to move this back how it was, because we want to not use this code for the go-gather branch of the unit test, (unless you have a plan to untangle the logic).

Reply:
this has been moved back.